### DISCUSSION TITLE
conflicting implementations of trait diesel_async::RunQueryDsl<_> for type paginate_async::PaginatedQuery<_, _>

### DISCUSSION THREAD

**Discussion by UVUUUVUU** [weiznich/diesel_async#112] [Q&A]
hello, i have defined a new trait by diesel_async:
`use std::marker::PhantomData;
use diesel::mysql::Mysql;
use diesel::pg::Pg;
use diesel::query_dsl::LoadQuery;
use diesel::sql_types::BigInt;
use diesel::{query_builder::*, QueryResult};
use diesel_async::RunQueryDsl;
use diesel_async::AsyncConnection;
pub trait Paginate: Sized {
fn paginate(self, offset: usize, limit: usize) -> PaginatedQuery<Self, DB> {
PaginatedQuery {
query: self,
offset: offset as i64,
limit: limit as i64,
_marker: PhantomData,
}
}
}
impl<T, DB> Paginate for T {}
#[derive(Debug)]
pub struct PaginatedQuery<T, Conn> {
query: T,
offset: i64,
limit: i64,
_marker: PhantomData,
}

impl<T, Conn> PaginatedQuery<T, Conn> {
pub fn load_and_total<'a, U>(self, conn: &mut Conn) -> QueryResult<(Vec, i64)>
where
Self: LoadQuery<'a, Conn, (U, i64)>,
{
let results = self.load::<(U, i64)>(conn)?;
let total = *results.get(0).map(|(, total)| total).unwrap_or(&0);
let records: Vec = results.into_iter().map(|(record, )| record).collect();
Ok((records, total))
}
}`
but the follow code reports an error.
`impl<T, Conn> RunQueryDsl for PaginatedQuery<T, Conn> {}`
the error is "conflicting implementations of traitdiesel_async::RunQueryDsl<>for typepaginate_async::PaginatedQuery<, _>conflicting implementation in cratediesel_async:
impl<T, Conn> diesel_async::RunQueryDsl for T;"
what should i do for this error? thank you.

**Comment by weiznich** [comment]
Thanks for writing this bug report. Please complete the issue template. Its mandatory for issues in our tracker.. If you don't feel that your report fits into the template consider moving it to discussion forum.

**Comment by weiznich** [comment]
I've now converted this issue into a discussion as it does not seem to be a bugreport at all, but rather a support question.
If you are looking for help here you need to:

* Fix the formatting of your post
* Include information about your diesel, diesel_async, and rustc versions
* Include the complete error message as emitted by rustc (all error messages)

Otherwise it's just not possible to help you in any way.

**Reply by UVUUUVUU** [reply]
ok, got it. thanks
